proxy/HostStatus.h

```diff
@@ -182,11 +218,16 @@ struct HostStatus {
   }
   void setHostStatus(const std::string_view name, const TSHostStatus status, const unsigned int down_time,
                      const unsigned int reason);
-  HostStatRec *getHostStatus(const std::string_view name);
-  void createHostStat(const std::string_view name, const char *data = nullptr);
-  void loadHostStatusFromStats();
+  void loadFromPersistentStore();
   void loadRecord(std::string_view name, HostStatRec &h);
-  RecErrT getHostStat(std::string &stat_name, char *buf, unsigned int buf_len);
+  HostStatRec *getHostStatus(const std::string_view name);
+  void getAllHostStatuses(std::vector<HostStatuses> &hosts);
```

Comment:
You could return a vector, it should be efficient due to Return Value Optimization.

Reply:
We should let the people who went to grad school tell us if it's wrong.